### PR TITLE
[ntuple] Rename 'unsplit field' to 'streamer field'

### PR DESCRIPTION
**Pull request by jblomer**
Including renaming of the dictionary option. As discussed with CMS, at this point we do not need a migration path from the old dictionary option names to the new ones.

@Dr15Jones FYI.

### CODE REVIEW

File: tree/ntuple/v7/test/rfield_streamer.cxx [hunk 1]
```diff
@@ -185,7 +185,7 @@ TEST(RField, UnsplitMerge)
    }
 
    // Now merge the inputs
-   FileRaii fileGuard3("test_ntuple_merge_unsplit_out.root");
+   FileRaii fileGuard3("test_ntuple_merge_streamer_out.root");
```
Comment (pcanal) [inline tree/ntuple/v7/test/rfield_streamer.cxx:188]: or?
```suggestion
   FileRaii fileGuard3("test_ntuple_merge_streamed_out.root");
```

**Comment by silverweed** [comment]
It looks to me like we are sometimes calling it "streamed" and sometimes "streamer": should we get rid of this discrepancy?

**Comment by jblomer** [comment]
We didn't want to go for `StreamedField` because it may sound to close to streaming in the sense of XRootD streaming. I think once we are deep in the serialization context, "streamed" has not that ambiguity anymore. If we want to avoid "streamed", we have to go for a longer name, e.g. `rntplStreamerMode(true)`.

**Comment by jblomer** [comment]
@silverweed: Here is an attempt with "streamer mode" instead of "streamed".

**Review by silverweed — APPROVED** [review]

**Review by pcanal — APPROVED** [review]

**Pull request merged**